addons/point_of_sale/__manifest__.py

```diff
@@ -66,119 +118,32 @@
-            'point_of_sale/static/src/js/ChromeWidgets/CashierName.js',
-            'point_of_sale/static/src/js/ChromeWidgets/ProxyStatus.js',
-            'point_of_sale/static/src/js/ChromeWidgets/SyncNotification.js',
-            'point_of_sale/static/src/js/ChromeWidgets/HeaderButton.js',
-            'point_of_sale/static/src/js/ChromeWidgets/SaleDetailsButton.js',
-            'point_of_sale/static/src/js/ChromeWidgets/CashMoveButton.js',
-            'point_of_sale/static/src/js/ChromeWidgets/TicketButton.js',
-            'point_of_sale/static/src/js/Misc/CurrencyAmount.js',
-            'point_of_sale/static/src/js/Misc/Draggable.js',
-            'point_of_sale/static/src/js/Misc/NotificationSound.js',
-            'point_of_sale/static/src/js/Misc/IndependentToOrderScreen.js',
-            'point_of_sale/static/src/js/Misc/AbstractReceiptScreen.js',
-            'point_of_sale/static/src/js/Misc/SearchBar.js',
-            'point_of_sale/static/src/js/ChromeWidgets/DebugWidget.js',
-            'point_of_sale/static/src/js/Popups/AbstractAwaitablePopup.js',
-            'point_of_sale/static/src/js/Popups/ErrorPopup.js',
-            'point_of_sale/static/src/js/Popups/ErrorBarcodePopup.js',
-            'point_of_sale/static/src/js/Popups/ConfirmPopup.js',
-            'point_of_sale/static/src/js/Popups/CashMovePopup.js',
-            'point_of_sale/static/src/js/Popups/ControlButtonPopup.js',
-            'point_of_sale/static/src/js/Popups/TextInputPopup.js',
-            'point_of_sale/static/src/js/Popups/TextAreaPopup.js',
-            'point_of_sale/static/src/js/Popups/ErrorTracebackPopup.js',
-            'point_of_sale/static/src/js/Popups/SelectionPopup.js',
-            'point_of_sale/static/src/js/Popups/EditListInput.js',
-            'point_of_sale/static/src/js/Popups/EditListPopup.js',
-            'point_of_sale/static/src/js/Popups/NumberPopup.js',
-            'point_of_sale/static/src/js/Popups/OfflineErrorPopup.js',
-            'point_of_sale/static/src/js/Popups/OrderImportPopup.js',
-            'point_of_sale/static/src/js/Popups/ProductConfiguratorPopup.js',
-            'point_of_sale/static/src/js/Popups/CashOpeningPopup.js',
-            'point_of_sale/static/src/js/Popups/ProductInfoPopup.js',
-            'point_of_sale/static/src/js/Screens/ProductScreen/ControlButtons/OrderlineCustomerNoteButton.js',
-            'point_of_sale/static/src/js/Popups/ClosePosPopup.js',
-            'point_of_sale/static/src/js/Popups/MoneyDetailsPopup.js',
+
+            # We declare these components first because they are referred in some components.
+            # We do this to get rid of the console warning message when opening the POS UI.
+            # TODO: Remove this special case by removing `ControlButtonsMixin`. It's better
+            #   to NOT rely on the order of loading of the js assets.
             'point_of_sale/static/src/js/Screens/ProductScreen/ControlButtons/SetPricelistButton.js',
```

Comment:
these two are probably not needed

Reply:
I removed it here: https://github.com/odoo/odoo/pull/82461/commits/2041439c6f96c4946e4e86c9554e4950a669202e

The changes are needed because:
1. Before it, whenever the control buttons are added, we automatically build the ordered list of control buttons with the assumption that the control button being added has reference control button that is already in the list.
2. Now that we remove the explicit declaration of the 2 control buttons, the other control buttons that depends on their location won't be able to find them, specially if the name of the dependent is "greater" than the parent control button. To be able to keep the control buttons ordering, we sort them at the first instantiation of the component. Succeeding rendering of the component uses the cached and ordered list of control buttons.